src/framework/MOM_intrinsic_functions.F90

```diff
@@ -25,4 +28,127 @@ function invcosh(x)
+  integer :: ex_3 ! One third of the exponent part of x, used to rescale x to get a.
+  integer :: itt
+
+  if ((x >= 0.0) .eqv. (x <= 0.0)) then
+    ! Return 0 for an input of 0, or NaN for a NaN input.
+    root = x
+  else
+    ex_3 = ceiling(exponent(x) / 3.)
+    ! Here asx is in the range of 0.125 <= asx < 1.0
+    asx = scale(abs(x), -3*ex_3)
+
+    ! This first estimate is one iteration of Newton's method with a starting guess of 1.  It is
+    ! always an over-estimate of the true solution, but it is a good approximation for asx near 1.
+    num = (2.0 + asx)
+    den = 3.0
+    ! Iteratively determine Root = asx**1/3 using Newton's method, noting that in this case Newton's
+    ! method converges monotonically from above and needs no bounding.  For the range of asx from
+    ! 0.125 to 1.0 with the first guess used above, 6 iterations suffice to converge to roundoff.
+    do itt=1,9
+      ! Newton's method iterates estimates as Root = Root - (Root**3 - asx) / (3.0 * Root**2), or
+      ! equivalently as Root = (2.0*Root**2 + asx) / (3.0 * Root**2).
+      ! Keeping the estimates in a fractional form Root = num / den allows this calculation with
+      ! fewer (or no) real divisions during the iterations before doing a single real division
+      ! at the end, and it is therefore more computationally efficient.
+
+      ! Because successive estimates of the numerator and denominator tend to be the cube of their
+      ! predecessors, the numerator and denominator need to be rescaled by division when they get
+      ! too large or small to avoid overflow or underflow in the convergence test below.  Were
+      ! this being done for 32 bit reals, the values to compare with would be about 1.0e9 and 1.0e-9.
+      if ((den > 1.0e75) .or. (den < 1.0e-75)) then
+        num = num / den ; den = 1.0
+      endif
+
+      num_prev = num ; den_prev = den
+      num = (2.0 * num_prev**3 + asx * den_prev**3)
+      den = 3.0 * (den_prev * num_prev**2)
+
+      !   If successive estimates of root are identical, this is an exact or converged solution.
+      if ((num * den_prev == num_prev * den) .or. (itt == 9)) then
+        root_asx = num / den
+        exit
+      endif
+
+      !   If the estimates are increasing, this also indicates convergence, but for a more subtle
+      ! reason.  Because Newton's method converges monotonically from above (at least for infinite
+      ! precision math), the only reason why this estimate could increase is if the iterations
+      ! have converged to a roundoff-level limit cycle around an irrational or otherwise
+      ! unrepresentable solution, with values only changing in the last bit or two.  If so, we
+      ! should stop iterating and accept the one of the current or previous solutions, both of
+      ! which will be within numerical roundoff of the true solution.
+      if (num * den_prev > num_prev * den) then
+        root_asx = num / den
```

Comment:
This could be written as an `if-else` block, which would avoid a potentially redundant division.

Reply:
Thank you for this suggestion; it has been implemented.